source/Calamari.Tests/KubernetesFixtures/KubernetesContextScriptWrapperLiveFixtureBase.cs

```diff
@@ -42,42 +52,98 @@ public void Setup()
 
-            Log = new DoNotDoubleLog();
+            redactionLog = new RedactedValuesLogger(Log = new DoNotDoubleLog());
 
             SetTestClusterVariables();
         }
 
-        protected KubernetesContextScriptWrapper CreateWrapper()
+        protected KubernetesContextScriptWrapper CreateWrapper(ICalamariFileSystem fileSystem = null)
         {
-            return new KubernetesContextScriptWrapper(variables, Log, new AssemblyEmbeddedResources(), new TestCalamariPhysicalFileSystem());
+            return new KubernetesContextScriptWrapper(variables, redactionLog, new AssemblyEmbeddedResources(), fileSystem ?? new TestCalamariPhysicalFileSystem());
         }
 
         void SetTestClusterVariables()
         {
 
-            variables.Set(SpecialVariables.Namespace, TestNamespace);
+            variables.Set(KubernetesSpecialVariables.Namespace, TestNamespace);
             variables.Set(ScriptVariables.Syntax, CalamariEnvironment.IsRunningOnWindows ? ScriptSyntax.PowerShell.ToString() : ScriptSyntax.Bash.ToString());
         }
 
-        CalamariResult ExecuteScript(IScriptWrapper wrapper, string scriptName)
+        CalamariResult ExecuteScript(IScriptWrapper wrapper, string scriptName) =>
+            ExecuteScript(new[] { wrapper }, scriptName);
+
+        CalamariResult ExecuteScript(IReadOnlyList<IScriptWrapper> additionalWrappers, string scriptName, ICalamariFileSystem fileSystem = null)
         {
-            var calamariResult = ExecuteScriptInternal(new CommandLineRunner(Log, variables), wrapper, scriptName);
+            var wrappers = new List<IScriptWrapper>(additionalWrappers);
+            if (variables.Get(SpecialVariables.Account.AccountType) == "AmazonWebServicesAccount")
+            {
+                wrappers.Add(new AwsScriptWrapper(redactionLog, variables));
+            }
+
+            var result = fileSystem != null
+                ? ExecuteWithRunScriptCommand(fileSystem, wrappers)
+                : ExecuteDirectlyWithScriptEngine(wrappers, scriptName);
 
             WriteLogMessagesToTestOutput();
 
-            return calamariResult;
+            return result;
         }
 
-        CalamariResult ExecuteScriptInternal(ICommandLineRunner runner, IScriptWrapper wrapper, string scriptName)
+        CalamariResult ExecuteWithRunScriptCommand(ICalamariFileSystem fileSystem, IEnumerable<IScriptWrapper> scriptWrappers)
         {
-            var wrappers = new List<IScriptWrapper>(new[] { wrapper });
-            if (variables.Get(Deployment.SpecialVariables.Account.AccountType) == "AmazonWebServicesAccount")
+            var oldEnvVars = new Dictionary<string, string>();
+            foreach (var envVar in GetEnvironments())
             {
-                wrappers.Add(new AwsScriptWrapper(Log, variables));
+                oldEnvVars[envVar.Key] = Environment.GetEnvironmentVariable(envVar.Key);
+                Environment.SetEnvironmentVariable(envVar.Key, envVar.Value);
```

Comment:
Can we do this without needing to modify the environment variables?  My concern that this could interfere with other tests, either when running concurrently, or if this fails before the variables are reset to their old values.

Reply:
This is fixed now! I had to adjust a few bits and pieces to be able to pass in the environment variables but now it works so I don't have to set them on the process 👍

Reply:
I'm just waiting for the build to come good...